src/server.c

```diff
@@ -146,6 +148,26 @@ char *COMPONENT_SEPARATOR = ";";
 
 const char *DEFAULT_USER_ID = "anonymous";
 
+const char *qd_log_message_components[] =
```

Comment:
You should move this out of server.c where it doesn't really belong.  Consider adding a message_trace.[ch] module for this logic.